corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js

```diff
@@ -43,13 +43,13 @@ hqDefine("data_dictionary/js/data_dictionary", [
 
                 for (let prop of group.properties) {
                     const isGeoCaseProp = (self.geoCaseProp === prop.name && prop.data_type === 'gps');
-                    if (prop.contains_case_data) {
+                    if (!prop.is_safe_to_delete) {
```

Comment:
This would be the same, but I wanted to make it explicit that we are marking the case type as unsafe to delete. This conditional check is also required as without it the last looped case property would get to determine whether a case type is safe to delete or not.

Reply:
> last looped case property would get to determine whether a case type is safe to delete or not.

Aha. I didn't notice that.
I guess its nicer to just extract that of the loop then to avoid confusion and use
`self.canDelete = !_.every(group.properties, function(prop) { return prop.is_safe_to_delete} )`

I understand if you wish to avoid going through the properties again though I believe the advantage of extracting this out are good to consider.

Reply:
For performance reasons, it's good to try and loop through the array of case properties as few times as possible. This is largely because the entire DD gets loaded on the initial page load and so if there exists a lot of case properties across multiple case types this could further hamper the loading performance of the DD. I think if we were able to incrementally load the DD then this would be much less of a problem.

@mkangia Would a comment here help to remove the confusion? Alternatively, we could also implement something as follows:
```
let unsafePropCount = 0;
for (let group of groupData) {
    ...
    for (let prop of group.properties) {
        if (!prop.is_safe_to_delete) {
            unsafePropCount += 1;
        }
        ...
    }
}
self.canDelete = Boolean(unsafePropCount);
```

Reply:
okay makes sense. 
Good to avoid the reiteration.

Oh, that count change looks an overkill.

How about
```
if (!prop.is_safe_to_delete && self.self.canDelete) {
    self.canDelete = false;
}
```
or else a comment is fine as well.
Thank you!

Reply:
That can work quite nicely. I'll implement the suggested code change, thanks!

Reply:
[406320d](https://github.com/dimagi/commcare-hq/pull/34073/commits/406320db66b4811c882aa86d79c3323f9b358ff5)